src/Metadata/DateTimeOptions.php

```diff
@@ -43,17 +44,30 @@ public function getZone(): ?string
         return $this->zone;
     }
 
+    /**
+     * @deprecated Please use {@see getDeserializeFormats}
+     */
     public function getDeserializeFormat(): ?string
     {
-        return $this->deserializeFormat;
+        foreach ($this->deserializeFormats ?? [] as $format) {
```

Comment:
we could fallback on assignment to `[]` instaed of null, then this becomes simpler.